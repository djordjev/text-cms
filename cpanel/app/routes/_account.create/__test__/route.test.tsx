import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import Route from '../route';

test('Route', () => {
  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/',
        Component: () => <Route />,
        action: () => {
          return {};
        }
      }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(screen.getByRole('textbox', { name: 'username' })).toBeInTheDocument();
  expect(screen.getByRole('button', { name: 'Sign Up' })).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
