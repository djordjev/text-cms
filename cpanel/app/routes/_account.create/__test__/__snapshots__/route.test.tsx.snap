// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Route 1`] = `
NodeList [
  <div>
    <h1
      class="u-uppercase u-tracking-wide u-font-bold u-text-center u-mb-3x"
    >
      Create a new account
    </h1>
    <form
      action="/"
      class="u-mb-3xs"
      method="post"
    >
      <label
        class="u-input u-input-bordered u-flex u-items-center u-gap-2 u-mb-3xs"
      >
        <svg
          aria-label="username"
          class="u-mr-2x"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 7a2 2 0 0 1 2 -2h14a2 2 0 0 1 2 2v10a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-10z"
          />
          <path
            d="M3 7l9 6l9 -6"
          />
        </svg>
        <input
          class="u-grow"
          name="username"
          placeholder="Username"
          type="text"
        />
      </label>
      <label
        class="u-input u-input-bordered u-flex u-items-center u-gap-2 u-mb-3xs"
      >
        <svg
          aria-label="password"
          class="u-mr-2x"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.555 3.843l3.602 3.602a2.877 2.877 0 0 1 0 4.069l-2.643 2.643a2.877 2.877 0 0 1 -4.069 0l-.301 -.301l-6.558 6.558a2 2 0 0 1 -1.239 .578l-.175 .008h-1.172a1 1 0 0 1 -.993 -.883l-.007 -.117v-1.172a2 2 0 0 1 .467 -1.284l.119 -.13l.414 -.414h2v-2h2v-2l2.144 -2.144l-.301 -.301a2.877 2.877 0 0 1 0 -4.069l2.643 -2.643a2.877 2.877 0 0 1 4.069 0z"
          />
          <path
            d="M15 9h.01"
          />
        </svg>
        <input
          class="u-grow"
          name="password"
          placeholder="Password"
          type="password"
        />
      </label>
      <label
        class="u-input u-input-bordered u-flex u-items-center u-gap-2 u-mb-3xs"
      >
        <svg
          aria-label="repeat password"
          class="u-mr-2x"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.555 3.843l3.602 3.602a2.877 2.877 0 0 1 0 4.069l-2.643 2.643a2.877 2.877 0 0 1 -4.069 0l-.301 -.301l-6.558 6.558a2 2 0 0 1 -1.239 .578l-.175 .008h-1.172a1 1 0 0 1 -.993 -.883l-.007 -.117v-1.172a2 2 0 0 1 .467 -1.284l.119 -.13l.414 -.414h2v-2h2v-2l2.144 -2.144l-.301 -.301a2.877 2.877 0 0 1 0 -4.069l2.643 -2.643a2.877 2.877 0 0 1 4.069 0z"
          />
          <path
            d="M15 9h.01"
          />
        </svg>
        <input
          class="u-grow"
          name="passwordRepeat"
          placeholder="Repeat password"
          type="password"
        />
      </label>
      <button
        class="u-btn u-btn-outline u-w-full u-uppercase"
      >
        Sign Up
      </button>
    </form>
    <p>
      Already have an account? Log in
       
      <a
        class="u-link"
        href="/login"
      >
        here
      </a>
    </p>
  </div>,
]
`;
